### PR TITLE
Create table user status type 35

### LINKED ISSUE #35: Create Table: user_status_type
### Overview

We need to create the `user_status_type` table so that we can update a shared data store across hackforla.org, vrms, civictechjobs, and tables (onboarding) project.

#### Details

A table and a model are the same thing

### Action Items

- [x] identify and document table description (see spreadsheet under Resources)
  - [ ] if not, reach out to PD leads
- [x] compare and check off the data fields (below) against the ERD. Note any fields in the ERD that's missing from the list. (see Resources)
- [x] compare the check off associated tables (below) against the ERD. Note any associated tables in the ERD that's missing from the list (see Resources)
- [x] create a single model in Django (defining schema)
- [x] add any seed data
- [x] write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project).
- [x] write an API end point
- [x] write API unit tests
- [x] document the endpoint
- [x] remove the block in issue #24
- [x] remove the dependencies and pull these out of the ice box: 
   - [x] #36 
   - [ ] PR #141
   - [x] #24

### Resources/Instructions

- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint
- See the [wiki page](https://github.com/hackforla/peopledepot/wiki/Create-Table-issues-data-gathering-workflow) for instructions.

### Description

User Status

### Data Fields

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] (PK) id - int - Record Id
   - [x] name - varchar
   - [x] description - varchar

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### Associated Tables

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] user (one-to-many)

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### PR DESCRIPTION
**Pull request by del9ra**
Fixes #35

### What changes did you make?

- created a model
- added seed data
- tested API endpoints and model

### Why did you make the changes (we will use this info to test)?
- to update a shared data store across hackforla.org, vrms, civictechjobs, and tables project

### CODE REVIEW

File: app/core/models.py [hunk 1]
```diff
@@ -426,3 +426,11 @@ class SocMajor(AbstractBaseModel):
 
     def __str__(self):
         return self.title
+
+
+class UserStatusType(AbstractBaseModel):
```
Comment (dmartin4820) [inline app/core/models.py:431]: Now that this model is going to be implemented, we should uncomment the one-to-many relationship [here](https://github.com/hackforla/peopledepot/blob/53b4a22507d040a9af5cc45a378b48afb01d8e79/app/core/models.py#L77), according to the tool tip under [here](https://github.com/hackforla/peopledepot/blob/53b4a22507d040a9af5cc45a378b48afb01d8e79/app/core/models.py#L77).

**Review by dmartin4820 — CHANGES_REQUESTED** [review]
Left some minor comments. I switched to your branch in my local peopledepot git area, and I ran 

```
1. ./scripts/buildrun.sh
2. ./scripts/test.sh
```

without issue.

Other than the few comments, it looks good to me 👍 .

**Review by fyliu — CHANGES_REQUESTED** [review]
Great work!
I just have a couple of small things.

**Review by dmartin4820 — APPROVED** [review]
No other changes from me. Tested buildrun and test scripts with no issues as well. Good work.

**Review by fyliu — APPROVED** [review]
This looks great! Thank you for adding the user status type model!

**Review by ethanstrominger — APPROVED** [review]

**Comment by fyliu** [comment]
I decided to squash the commits before merging because there's migrations that's replaced in later commits. Squashing them removes those intermediate files from history so that rolling forward from any past commit would work.

**Pull request merged**